admin/src/hooks/useReportsActionsHandlers.ts

```diff
@@ -0,0 +1,143 @@
+// @ts-nocheck
+
+import React, { useCallback } from "react";
+
+import { useMutation, useQueryClient } from "react-query";
+import { useOverlayBlocker, useNotification } from "@strapi/helper-plugin";
+import { handleAPIError } from "../utils";
+import {
+  blockItem,
+  blockItemThread,
+  resolveAllAbuseReportsForComment,
+  resolveAllAbuseReportsForThread,
+  unblockItem,
+  unblockItemThread,
+} from "../pages/utils/api";
+
+const useReportsActionsHandlers = (
+  allowedActions,
+  commentId,
+  mutation,
+  onSelectionChange,
+  reports,
+  selectedReports,
+  updateReports,
+) => {
+  const { canModerate, canReviewReports } = allowedActions;
+
+  const { lockApp, unlockApp } = useOverlayBlocker();
+
+  const queryClient = useQueryClient();
+  const toggleNotification = useNotification();
+
+  const onSuccess =
+    (stateAction = () => {}, indalidate = true) =>
+    async () => {
+      if (indalidate) {
+        await queryClient.invalidateQueries("get-data");
```

Comment:
IMO queries indexes/keys should be moved to some config space. We can work something out off-line.